js/runtime.js

```diff
@@ -472,24 +476,36 @@
                 _gaq.push(['_trackEvent', 'Blocks', 'Image', 'get']);
                 return assets.images[path];
             },
-            drawAtPoint: function(img, pt, w, h){
+            drawAtPoint: function(img, pt){
                 _gaq.push(['_trackEvent', 'Blocks', 'Image', 'drawAtPoint']);
-                ctx().drawImage(img, pt.x, pt.y, w, h);
+                img.drawAtPoint(ctx, pt);
             },
-            drawAtPointWithSize: function(img, pt, sz){
-                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'drawAtPointWithSize']);
-                ctx().drawImage(img, pt.x, pt.y, sz.w, sz.h);
+            setWidth: function(img, w){
+                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'setWidth']);
+                img.setWidth(w);
             },
-            getScaledAtOrigin: function(path, sz){
-                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'getScaledAtOrigin']);
-                var img = images[path];
-                return {
-                    name: 'Image',
-                    draw: function draw(inner_ctx){
-                        inner_ctx.drawImage(img, -sz.w/2, -sz.h/2, sz.w, sz.h);
-                    }
-                };
+            setHeight: function(img, h){
+                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'setHeight']);
+                img.setHeight(h);
+            },
+            setSize: function(img, sz){
+                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'setSize']);
+                img.setSize(sz);
+            },
+            scale: function(img, scaleFactor){
+                _gaq.push(['_trackEvent', 'Blocks', 'Image', 'scale']);
+                img.scale(scaleFactor);
             }
+            // getScaledAtOrigin: function(path, sz){
```

Comment:
Good question. I think that was an approach I tried, then tried something else and forgot to go back and delete? I'll remove it.